test/unittests/execute_test.cpp

```diff
@@ -1513,29 +1514,43 @@ TEST(execute, i32_xor)
 
 TEST(execute, i32_shl)
 {
-    const auto [trap, ret] = execute_binary_operation(Instr::i32_shl, 21, 1);
-
-    ASSERT_FALSE(trap);
-    ASSERT_EQ(ret.size(), 1);
-    EXPECT_EQ(ret[0], 42);
+    EXPECT_RESULT(execute_binary_operation(Instr::i32_shl, 21, 1), 42);
+    EXPECT_RESULT(execute_binary_operation(Instr::i32_shl, 0xffffffff, 0), 0xffffffff);
+    EXPECT_RESULT(execute_binary_operation(Instr::i32_shl, 0xffffffff, 1), 0xfffffffe);
+    EXPECT_RESULT(execute_binary_operation(Instr::i32_shl, 0xffffffff, 31), 0x80000000);
+    EXPECT_RESULT(execute_binary_operation(Instr::i32_shl, 0xffffffff, 32), 0xffffffff);
+    EXPECT_RESULT(execute_binary_operation(Instr::i32_shl, 0xffffffff, 33), 0xfffffffe);
+    EXPECT_RESULT(execute_binary_operation(Instr::i32_shl, 0xffffffff, 63), 0x80000000);
 }
 
 TEST(execute, i32_shr_s)
 {
-    const auto [trap, ret] = execute_binary_operation(Instr::i32_shr_s, uint64_t(-84), 1);
-
-    ASSERT_FALSE(trap);
-    ASSERT_EQ(ret.size(), 1);
-    EXPECT_EQ(ret[0], uint64_t(-42));
+    EXPECT_RESULT(execute_binary_operation(Instr::i32_shr_s, uint64_t(-84), 1), uint64_t(-42));
+    EXPECT_RESULT(execute_binary_operation(Instr::i32_shr_s, 0xffffffff, 0), int32_t(0xffffffff));
+    EXPECT_RESULT(execute_binary_operation(Instr::i32_shr_s, 0xffffffff, 1), int32_t(0xffffffff));
+    EXPECT_RESULT(execute_binary_operation(Instr::i32_shr_s, 0xffffffff, 31), int32_t(0xffffffff));
+    EXPECT_RESULT(execute_binary_operation(Instr::i32_shr_s, 0xffffffff, 32), int32_t(0xffffffff));
```

Comment:
Spectests also have cases like 0x80000000 >> 32 = 1

Reply:
Many others can be added, but I think it is fine to leave it as is until next bug.